api/src/main/java/org/openmrs/module/patientflags/db/hibernate/HibernateFlagDAO.java

```diff
@@ -168,7 +188,23 @@ public void purgeTag(Integer tagId) throws DAOException {
 	public Priority getPriority(Integer priorityId) {
 		return (Priority) sessionFactory.getCurrentSession().get(Priority.class, priorityId);
 	}
-	
+
+	public Priority getPriorityByUuid(String uuid) throws DAOException {
+		return (Priority) this.sessionFactory.getCurrentSession().createQuery("from Priority p where p.uuid = :uuid").setString("uuid", uuid).uniqueResult();
+	}
+
+	public Priority getPriorityByName(String name) throws DAOException {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Priority.class);
+		criteria.add(Restrictions.ilike("name", name, MatchMode.EXACT));
+
+		if (criteria.list().size() > 0) {
+			// note the assumption here is that two priorities with the same case-insensitive tags aren't allowed; if there are two, this just returns the first one it finds

```

Comment:
If more than one is not allowed then it is more correct to throw an exception rather than returning a random one...